torch/csrc/jit/serialization/export.cpp

```diff
@@ -954,6 +949,11 @@ export_onnx(
 
 std::string serialize_model_proto_to_string(
     const std::shared_ptr<::ONNX_NAMESPACE::ModelProto>& model_proto) {
+  const size_t proto_size = model_proto->ByteSizeLong();
```

Comment:
Before we do this check, we can check use_external_data_format is True or not. If it is True, we may simply skip this check.

Reply:
use_external_data_format is not accessible from here. Besides, this is an assertion type of check, to prevent potential segfault from the following call if the protobuf size should exceed 2GB. So i think it would be helpful to leave it here.